scripts/ingest-local-setup.sh

```diff
@@ -0,0 +1,36 @@
+#! /bin/bash
+
+# Called in docker-compose-setup.sh for Docker development environment
+#
+# Keep "Dev env vars" synced with `setup-mongo-dev.sh`
+
+VAULT_TOKEN_PATH="$1"
+if [[ -z "$VAULT_TOKEN_PATH" ]]
+then
+  echo "You must provide a path to a GitHub token to proceed, e.g. ~/.github-token"
+  exit 1
+fi
+vault login -method=github token=$(cat $VAULT_TOKEN_PATH)
+if [[ $? -ne 0 ]]
+then
+  echo "Unable to authenticate into Vault"
+  exit 1
+fi
+
+# Dev env vars
+BROAD_USER=`whoami`
+MONGODB_USERNAME='single_cell'
+DATABASE_NAME='single_cell_portal_development'
+MONGODB_PASSWORD=`vault read secret/kdux/scp/development/$BROAD_USER/mongo/user | grep password | awk '{ print $2 }' `
+DATABASE_HOST=`vault read secret/kdux/scp/development/$BROAD_USER/mongo/hostname | grep ip | awk '{ $2=substr($2,2,length($2)-2); print $2 }' ` 
+BYPASS_MONGO_WRITES='yes'
+BARD_HOST_URL="https://terra-bard-dev.appspot.com"
+
+mkdir -p config
+echo "export BROAD_USER=${BROAD_USER}" >| config/.ingest_env
+echo "export MONGODB_USERNAME=${MONGODB_USERNAME}" >> config/.ingest_env
+echo "export DATABASE_NAME=${DATABASE_NAME}" >> config/.ingest_env
+echo "export MONGODB_PASSWORD=${MONGODB_PASSWORD}" >> config/.ingest_env
+echo "export DATABASE_HOST=${DATABASE_HOST}" >> config/.ingest_env
+echo "export BYPASS_MONGO_WRITES=${BYPASS_MONGO_WRITES}" >> config/.ingest_env
+echo "export BARD_HOST_URL=${BARD_HOST_URL}" >> config/.ingest_env
```

Comment:
Does `config/.ingest_env` hang around on my filesystem? ie. database password in a plaintext file that is only protected by virtue of it being a dotfile (ie. not listed by default)? or am I overlooking a step in a later script that deletes this file after it is used? Or is it referenced later by other scripts and needs to be kept - if it needs to be kept, should we `chmod 600` the file? (perhaps this caution is overkill for a file on a local developer laptop because `group` and `world` access isn't as relevant as on a shared environment?)

Reply:
That's a great point - it does persist on your machine.  We have the same behavior in `single_cell_portal_core`.  The "problem" is that we can't run any tasks automatically when you `CTRL^C` or type `exit` to kill the service.  We could add that step to a `docker-compose-cleanup.sh` script, but I agree that it's probably "caution overkill" since none of these secrets are for a deployed environment.

Reply:
Okay - as long as we've thought this all through (forgot this is just the creds for my local instance).

Reply:
Further detail on this good question seems worthwhile.  

In addition to A) reusing that previously-approved pattern Jon notes, we also mitigate security issues with these secrets by B) automatically excluding `config` from Git via [the new `.gitignore` entry](https://github.com/broadinstitute/scp-ingest-pipeline/pull/290/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947R3), and C) having our machines configured with `git-secrets` (completely [handled via BITS](https://broadinstitute.slack.com/archives/CADU7L0SZ/p1674575035693219) for recent-ish new machines).